observer.go

```diff
@@ -6,6 +6,8 @@ import (
 	"sync"
 	"sync/atomic"
 	"time"
+
+	"github.com/hashicorp/go-multierror"
```

Comment:
It seems that the dependence is missing.

```log
Check failure on line 10 in observer.go
GitHub Actions / unit tests
no required module provides package github.com/hashicorp/go-multierror; to add it:
```

Reply:
sorry, forgot to go mod tidy again. Fixed.